javaslang/src/main/java/javaslang/collection/BitMappedTrie.java

```diff
@@ -345,6 +345,18 @@ private int filter(Predicate<? super T> predicate, Object results, int index, T
         return index;
     }
 
+    Object[] toArray() {
+        final Object[] results = new Object[length];
+        visit((index, leaf, start, end) -> {
+            final int copied = end - start;
+            for (int i = start; i < end; i++) {
+                results[index + i] = type.getAt(leaf, i);
```

Comment:
@paplorinc Maybe we could benefit from both ideas, having an iterator() and a visitor().

We observe that an iterator() is not as performant as a visitor (because of the additional method calls, see Vector).

What if we add an additional method to Value, say `void visit(BiFunction<T, Integer>)`. I don't know which return type makes most sense here, but I want to mimic the internal Vector visit method.

Given that some of the existing collection methods might be implemented in an easier way. Currently I have no example other than changing an array:

```java
interface Value<T> extends Iterable<T> {
    
    default Object[] toJavaArray() {
        final Object[] results = new Object[length];
        visit((value, index) -> results[index] = value);
        return results;
    }

}
```

**Notes:** I know that an index does not make sense for all collections, e.g. HashMap and HashSet. But for Sorted* and Linked* collections it does make sense, so all Traversables should have that method.

If the conversion methods may benefit from the visit method (like toArray()), then Value should have that method, even if it is trivial for single-valued types like Option (beside the empty case).

Looks like a nice addition to me.... what do you think?